lib/urn.js

```diff
@@ -44,7 +44,77 @@ module.exports = {
       if (!urn.match(allowedProtocolsRegExp) || !urn.match(/^([^:]+:)+[^:]+$/)) {
         return false;
       }
+
+      // Parse ID portion of URN
+      var cat = urn.split(':').slice(0, -1)
+      var id = urn.split(':').slice(-1)[0];
+
+      // Split identity by each byte
+      var charArr = id.match(/.{1,2}/g);
+
+      // Match category of URN
+      switch (cat) {
+        case 'pbk:ec:secp256r1':
+          // Check for compressed/uncompressed ECDSA public keys
+          if (charArr[0] === '02' || charArr[0] === '03') {
+            // Check length of compressed public key
+            if (charArr.length !== 25) {
```

Comment:
Right but the number of characters being fed in is 50, and given we parse that by every two characters we validate on an array of size 25. The user will be inputting a compressed public key as such `0260fed4ba255a9d31c961eb74c6356d68c049b8923b61fa6ce669622e60f29fb6`, so it makes sense to validate directly on this.